accounting/aggregations/create_report.py

```diff
@@ -0,0 +1,214 @@
+
+    email_opts = parser.add_argument_group("email options")
+    for name, props in EMAIL_ARGS.items():
+        email_opts.add_argument(name, **props)
+
+    es_opts = parser.add_argument_group("elasticsearch options")
+    for name, props in ELASTICSEARCH_ARGS.items():
+        es_opts.add_argument(name, **props)
+
+    output_opts = parser.add_argument_group("output options")
+    for name, props in OUTPUT_ARGS.items():
+        output_opts.add_argument(name, **props)
+
+    return parser.parse_args()
+
+def get_client(args: dict) -> elasticsearch.Elasticsearch:
+    # set up elasticsearch options
+    es_opts = {
+        "timeout" : 120,
+        "hosts" : [args["es_url_prefix"] + args["es_host"]], 
+    }
+
+    if args["es_use_https"]:
+        if args["es_ca_certs"]:
+            es_opts.update({"ca_certs" : str(args["es_ca_certs"])})
+        
+        elif importlib.util.find_spec("certifi") is None:
+            print("error: using HTTPS requires either ca_certs or the certifi library to be installed")
+            sys.exit(1)
+
+        es_opts.update({"hosts" : "https://" + args["es_host"]})
+        es_opts.update({"use_https" : True})
+        es_opts.update({"verify_certs" : True}) 
+
+    if (not args["es_user"]) ^ (not args["es_password_file"]):
+        print("error: you must specify a username and password, or neither")
+        sys.exit(1)
+
+    elif args["es_user"] and args["es_password_file"]:
+        passwd_str = ""    
+        with open(args["es_password_file"], 'r') as file:
+           passwd_str = file.read()
+ 
+        es_opts.update({"http_auth" : (args["es_user"], passwd_str)})
+
+    client = elasticsearch.Elasticsearch(**es_opts)
+    connections.create_connection(alias="default", client=client)
+   
+    return client
+
+def main():
+    # parse arguments
+    args = parse_args()
+
+    # read the config file if there is one
+    es_opts = {}
+    if args.es_config_file:
+        es_opts = json.load(args.es_config_file.open())
+    else:
+        es_opts = {arg: v for arg, v in vars(args).items() if arg.startswith("es_")}
+  
+    client = get_client(es_opts)
+
+    # cast agg_by to a list if not already
+    if not isinstance(args.es_agg_by, list):
+        args.es_agg_by = [args.es_agg_by]
+     
+    results = []
+    for field in args.es_agg_by:
+        results.append((field, run_query(client, es_opts, args, field)))
+
+    # generate the outputs
+    if args.print_table: 
+        for title, r in results:
+            print('\n')            
+            print(f"CHTC Jobs by {title.split('.')[0]} for {args.start.strftime('%Y-%m-%d %H:%M:%S')} TO {args.end.strftime('%Y-%m-%d %H:%M:%S')}")
+            table(r)
+            
+    # send an email
+    html_tables = []
+    for title, r in results:
+        header_str = f"<h3>CHTC Jobs by {title.split('.')[0]} from {args.start.strftime('%Y-%m-%d %H:%M:%S')} TO {args.end.strftime('%Y-%m-%d %H:%M:%S')}</h3>"
```

Comment:
```suggestion
        header_str = f"<h1>CHTC Jobs by {title.split('.')[0]} from {args.start.strftime('%Y-%m-%d %H:%M:%S')} TO {args.end.strftime('%Y-%m-%d %H:%M:%S')}</h1>"
```
This is a little pedantic, feel free to push back here... I know _browsers_ turn `<h3>` into probably the size/weight that looks most pleasant here, but technically since there's no section above it, it should be `<h1>`. The font could always be adjusted for the `h1` elements  in the stylesheet below.

Reply:
On second thought, I definitely agree on this. Using h1 and changing the font size/weight with a stylesheet makes much more sense that trying to style with HTML